src/CabanaPD_Types.hpp

```diff
@@ -14,19 +14,49 @@
 
 namespace CabanaPD
 {
+// Mechanics types.
 struct Elastic
 {
 };
 struct Fracture
 {
 };
+
+// Thermal types.
+struct TemperatureIndependent
+{
+};
 struct TemperatureDependent
 {
 };
-struct TemperatureIndependent
+struct DynamicTemperature : public TemperatureDependent
+{
+    using base_type = TemperatureDependent;
+};
+
+//! Static type checkers.
+template <class>
+struct is_temperature_dependent : public std::false_type
```

Comment:
Could you please help me understand the different checks here Lines 38-55? I am trying to get the logic.

Reply:
Temperature dependent includes both with and without heat transfer, so we needed a compound check here